lib/matplotlib/axes/_axes.py

```diff
@@ -2465,10 +2487,16 @@ def broken_barh(self, xranges, yrange, **kwargs):
         self._process_unit_info(xdata=xdata,
                                 ydata=ydata,
                                 kwargs=kwargs)
-        xranges = self.convert_xunits(xranges)
+        xnew = []
+        for xr in xranges:
```

Comment:
```
for x, dx in xranges:
    x_conv = self.convert_xunits(x)
    dx_conv = self._convert_dx(...)
    xranges_conv.append((...))
```

(Note that your current implementation will silently work with xranges items with more than 2 elements, silently dropping the extra ones.)

Reply:
Even simpler with a list comprehension:
~~~
xranges_conv = [(self.convert_xunits(x),
                 self._convert_dx(x, x0, dx, self.convert_xunits))
                for x, dx in xranges]
~~~

Reply:
doesnt work I think? convert_dx needs the converted x.

Reply:
Ahem, OK, reverted...

@anntzer I don't see whats wrong w/ silently dropping extra stuff.  Is that worse than throwing an explicit but mysterious error?

Reply:
Because it's rather likely to hide a bug on the caller's side (if they're not passing in data in the right format, what's the chance taking the first two elements is going to yield the plot they intended?).

Reply:
OK, well then lets just check the length and throw a `ValueError` so we can control the message...

Reply:
if anything that error message should go to the BrokenBarHCollection constructor.

Reply:
As a user, I prefer errors to be raised in the routine I called rather than in something I had no idea existed...

Reply:
...  But we need the validation here, as you just pointed out above?  But you want to do it implictly, by trying to unpack the tuple, which would also yield a ValueError, but more unhelpfully: `ValueError: too many values to unpack (expected 2)`.  This is just providing a more helpful error.

Reply:
ah, fair enough

Reply:
.... not sure what that is and google wasn’t particularly helpful.  How would this get used in this case?